lib/ProductOpener/Display.pm

```diff
@@ -3325,8 +3323,7 @@ sub display_scatter_plot($$$) {
 				$data{url} = $url;
 				$data{img} = display_image_thumb($product_ref, 'front');
 				
-				defined $series{$seriesid} or $series{$seriesid} = '';
-				$series{$seriesid} .= encode_json(\%data) . ',';
+				$series{$seriesid} = \%data;
```

Comment:
I haven't tested the code, but it seems that $series{$seriesid} will only contain data for 1 product, instead of data for all products in the graphs, this is in a while loop that goes over all products returned by the search query. If you want to make only one call to JSON::PP->new->encode at the end, then I'm guessing we need to push the data values in an array, instead of using a scalar for $series{$seriesid}.

Reply:
Good point. I'll fix that later. Merci!